test/parallel/test-http2-pipe-named-pipe.js

```diff
@@ -15,19 +15,22 @@ const path = require('path');
 const tmpdir = require('../common/tmpdir');
 tmpdir.refresh();
 const loc = fixtures.path('person-large.jpg');
-const fn = path.join(tmpdir.path, 'http2-url-tests.js');
+const fn = path.join(tmpdir.path, 'person-large.jpg');
 
 const server = http2.createServer();
 
 server.on('stream', common.mustCall((stream) => {
   const dest = stream.pipe(fs.createWriteStream(fn));
 
-  dest.on('finish', () => {
-    assert.strictEqual(fs.readFileSync(loc).length,
-                       fs.readFileSync(fn).length);
-  });
-  stream.respond();
-  stream.end();
+  stream.on('end', common.mustCall(() => {
+    stream.respond();
+    stream.end();
+  }));
```

Comment:
I'm really confused by the fact that calling `stream.end()` after the `'end'` event is emitted on `stream` can have any effect.

Reply:
`stream.end()` is the writable side, `'end'` is on the readable side. They are not directly related.

Reply:
would it be possible to use `pipeline` somehow to simplify the test?

Reply:
It would but it does not make much sense in my opinion. We can call `stream.respond()` and `stream.end()` when the destination stream emits `'finish'` and not use the `'end'` event but I would prefer to do it as soon as possible.